libs/auth/scripts/program_production_system_administrator_java_card_without_vx_admin.ts

```diff
@@ -0,0 +1,83 @@
+import fs from 'fs/promises';
+import path from 'path';
+import { DirResult, dirSync } from 'tmp';
+import { extractErrorMessage } from '@votingworks/basics';
+
+import { PROD_VX_CERT_AUTHORITY_CERT_PATH } from '../src';
+import { CERT_EXPIRY_IN_DAYS, constructMachineCertSubject } from '../src/certs';
+import { createCert } from '../src/cryptography';
+import { getRequiredEnvVar } from '../src/env_vars';
+import { JavaCard } from '../src/java_card';
+import {
+  generatePrivateKey,
+  programSystemAdministratorJavaCard,
+} from './utils';
+
+const jurisdiction = getRequiredEnvVar('VX_MACHINE_JURISDICTION');
+const vxPrivateKeyPath = getRequiredEnvVar('VX_PRIVATE_KEY_PATH');
+
+async function instantiateJavaCardWithOneOffVxAdminPrivateKeyAndCertAuthorityCert(
+  tempDirectoryPath: string
+): Promise<JavaCard> {
+  const vxAdminPrivateKey = await generatePrivateKey();
+  const vxAdminPrivateKeyPath = path.join(
+    tempDirectoryPath,
+    'vx-admin-private-key.pem'
+  );
+  await fs.writeFile(vxAdminPrivateKeyPath, vxAdminPrivateKey);
+
+  const vxAdminCertAuthorityCert = await createCert({
+    certKeyInput: {
+      type: 'private',
+      key: { source: 'file', path: vxAdminPrivateKeyPath },
+    },
+    certSubject: constructMachineCertSubject('admin', jurisdiction),
+    certType: 'cert_authority_cert',
+    expiryInDays: CERT_EXPIRY_IN_DAYS.MACHINE_VX_CERT,
+    signingCertAuthorityCertPath: PROD_VX_CERT_AUTHORITY_CERT_PATH,
+    signingPrivateKey: { source: 'file', path: vxPrivateKeyPath },
+  });
+  const vxAdminCertAuthorityCertPath = path.join(
+    tempDirectoryPath,
+    'vx-admin-cert-authority-cert.pem'
+  );
+  await fs.writeFile(vxAdminCertAuthorityCertPath, vxAdminCertAuthorityCert);
+
+  const card = new JavaCard({
+    cardProgrammingConfig: {
+      vxAdminCertAuthorityCertPath,
+      vxAdminPrivateKey: { source: 'file', path: vxAdminPrivateKeyPath },
+    },
+    vxCertAuthorityCertPath: PROD_VX_CERT_AUTHORITY_CERT_PATH,
+  });
+
+  return card;
+}
+
+/**
+ * A script for programming a production system administrator Java Card without a VxAdmin. This is
+ * useful for preparing backup cards after we've shipped a VxAdmin.
+ */
+export async function main(): Promise<void> {
+  let exitCode: number = 0;
+  let tempDirectory: DirResult | undefined;
+  try {
+    tempDirectory = dirSync({ unsafeCleanup: true });
+    const card =
+      await instantiateJavaCardWithOneOffVxAdminPrivateKeyAndCertAuthorityCert(
+        tempDirectory.name
+      );
+    await programSystemAdministratorJavaCard({
+      card,
+      jurisdiction,
+      isProduction: true,
+    });
+  } catch (error) {
+    console.error(`❌ ${extractErrorMessage(error)}`);
+    exitCode = 1;
+  } finally {
+    tempDirectory?.removeCallback();
```

Comment:
is this doing the removal of tmp files?

Reply:
Yes exactly. And because we use `{ unsafeCleanup: true }` when creating the directory, we ensure that the files in it are deleted, too. Without that flag, `removeCallback` fails if the directory still contains files in it

Added a comment for clarity!